Client/BraveSkus/BraveSkusManager.swift

```diff
@@ -0,0 +1,154 @@
+      return
+    }
+    
+    Logger.module.debug("Refreshing sku credential. Clearing old credential from persistence.")
+    
+    BraveVPN.clearSkusCredentials()
+    
+    manager.credentialSummary(for: domain) { completion in
+      Logger.module.debug("credentialSummary response")
+    }
+  }
+  
+  // MARK: - Handling SKU methods.
+  
+  func refreshOrder(for orderId: String, domain: String, resultJSON: @escaping (Any?) -> Void) {
+    sku.refreshOrder(domain, orderId: orderId) { completion in
+      do {
+        guard let data = completion.data(using: .utf8) else { return }
+        let json = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed)
+        Logger.module.debug("refreshOrder json parsed successfully")
+        resultJSON(json)
+      } catch {
+        resultJSON(nil)
+        Logger.module.error("refrshOrder: Failed to decode json: \(error.localizedDescription)")
+      }
+    }
+  }
+  
+  func fetchOrderCredentials(for orderId: String, domain: String, resultCredential: @escaping (String) -> Void) {
+    sku.fetchOrderCredentials(domain, orderId: orderId) { completion in
+      Logger.module.debug("skus fetchOrderCredentials")
+      resultCredential(completion)
+    }
+  }
+  
+  func prepareCredentialsPresentation(for domain: String, path: String,
+                                      resultCredential: ((String) -> Void)?) {
+    Logger.module.debug("skus prepareCredentialsPresentation")
+    sku.prepareCredentialsPresentation(domain, path: path) { credential in
+      if !credential.isEmpty {
+        if let vpnCredential = BraveSkusWebHelper.fetchVPNCredential(credential, domain: domain) {
+          Preferences.VPN.skusCredential.value = credential
+          Preferences.VPN.skusCredentialDomain.value = domain
+          Preferences.VPN.skusCredentialExpirationDate.value = vpnCredential.expirationDate
+          
+          BraveVPN.setCustomVPNCredential(vpnCredential)
+        }
+      } else {
+        Logger.module.debug("skus empty credential from prepareCredentialsPresentation call")
+      }
+      
+      resultCredential?(credential)
+    }
+  }
+  
+  func credentialSummary(for domain: String, resultJSON: @escaping (Any?) -> Void) {
+    sku.credentialSummary(domain) { [weak self] completion in
+      do {
+        Logger.module.debug("skus credentialSummary")
+        
+        guard let data = completion.data(using: .utf8) else { return }
+        let json = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed)
+        
+        resultJSON(json)
+        
+        let jsonDecoder = JSONDecoder()
+        jsonDecoder.keyDecodingStrategy = .convertFromSnakeCase
+        let credentialSummaryJson = try jsonDecoder.decode(CredentialSummary.self, from: data)
+        
+        if credentialSummaryJson.isValid {
+          Preferences.VPN.expirationDate.value = credentialSummaryJson.expiresAt
```

Comment:
We don't want to use the date from credentialSummary since that's the end of the monthly term (ex: if it renews on the 15th, it would be December 15th). We want to store the date which comes when the credential is expired (the part from the cookie)

Reply:
Good find. I don't need this line, as I set the expiration date later in `prepareCredentialsPresentation` based off the cookie's date

Reply:
Update: I simplified it so there is a single stored expiration date for either the Apple's iAP or our SKU mechanism